### PR TITLE
Add spooled set stormtype, resolves SYN-7020, SYN-3401

### PR DESCRIPTION
**Pull request by rakuy0**
- Also fix some spelling and doc issues.

### CODE REVIEW

File: synapse/lib/stormlib/cortex.py [hunk 1]
```diff
@@ -1055,7 +1055,7 @@ class CortexHttpApi(s_stormtypes.Lib):
                       {'name': 'readonly', 'type': 'boolean',
                        'desc': 'Run the Extended HTTP Storm methods in readonly mode.', 'default': False},
                   ),
-                  'returns': {'type': 'http:api', 'desc': 'A new http:api object.'}}},
+                  'returns': {'type': 'http:api', 'desc': 'A new HTTP API object.'}}},
```
Comment (invisig0th) [inline synapse/lib/stormlib/cortex.py:1058]: The `http:api` object referenced here is  Synapse type :)

Comment (rakuy0) [reply synapse/lib/stormlib/cortex.py]: Yea, this just ends up being the weird URL that makes an errant link that goes to `http://api`. I tweaked it to be more in line with how the cron libs look.

**Review by invisig0th — DISMISSED** [review]

File: synapse/lib/stormlib/spooled.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    '''
+    _storm_typename = 'spooled:set'
+    _ismutable = True
+
+    def __init__(self, valu, path=None):
+
+        s_stormtypes.Prim.__init__(self, valu, path=path)
+        self.locls.update(self.getObjLocals())
+
+    async def iter(self):
```
[the remark this answers is not in the document]
Comment (rakuy0) [reply synapse/lib/stormlib/spooled.py:68]: Yes. The `self.valu` here is a synapse.lib.spooled.Set object, not a normal python set, and the Spooled Set doesn't implement an `__iter__` method, so we've gotta use the `__aiter__` method that the Spooled Set implements.

File: synapse/lib/stormlib/spooled.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    _storm_locals = (
+        {'name': 'set', 'desc': '''
+            Get a Spooled Storm Set object.
+
+            A Spooled Storm Set object is memory-safe to grow to extraordinarily large sizes,
+            as it will fallback to file backed storage, with two restrictions. First
+            is that all items in the set can be serialized to a file if the set grows too large,
+            so all items added must be a serializable Storm primitive. Second is that when an
+            item is added to the Set, because it could be immediately written disk,
+            do not hold any references to it outside of the Set itself,as the two objects could
```
Comment (Cisphyx) [inline synapse/lib/stormlib/spooled.py:20]:
```suggestion
            do not hold any references to it outside of the Set itself, as the two objects could
```

**Review by Cisphyx — APPROVED** [review]

**Review by vEpiphyte — APPROVED** [review]

File: synapse/lib/stormlib/cortex.py [hunk 4]
```diff
@@ -1055,24 +1055,24 @@ class CortexHttpApi(s_stormtypes.Lib):
...
+        {'name': 'get', 'desc': 'Get an Extended ``http:api`` object.',
          'type': {'type': 'function', '_funcname': 'getHttpApi',
                   'args': (
                       {'name': 'iden', 'type': 'string',
-                       'desc': 'The iden of the API to retreive.'},
+                       'desc': 'The iden of the API to retrieve.'},
                   ),
-                  'returns': {'type': 'http:api', 'desc': 'The http:api object.'}}},
-        {'name': 'list', 'desc': 'Get all the Extneded HTTP APIs on the Cortex',
+                  'returns': {'type': 'http:api', 'desc': 'The HTTP API object.'}}},
```
Comment (invisig0th) [inline synapse/lib/stormlib/cortex.py:1072]: missed one?

**Review by invisig0th — CHANGES_REQUESTED** [review]

File: synapse/lib/stormlib/cortex.py (hunk 4 above)
Comment (rakuy0) [reply synapse/lib/stormlib/cortex.py]: Yea, just missed one.

**Review by vEpiphyte — APPROVED** [review]

**Review by Cisphyx — APPROVED** [review]

**Review by invisig0th — APPROVED** [review]

**Pull request merged**